src/components/dialog.js

```diff
@@ -0,0 +1,102 @@
+(() => ({
+  name: 'Dialog',
+  type: 'CONTAINER_COMPONENT',
+  allowedTypes: ['BODY_COMPONENT', 'CONTENT_COMPONENT'],
+  orientation: 'HORIZONTAL',
+  jsx: (() => {
+    const { env } = B;
+    const { Dialog } = window.MaterialUI.Core;
+
+    const isEmpty = children.length === 0;
+    const isDev = env === 'dev';
+    const isPristine = isDev && isEmpty;
+    const isContentVisible = isDev && options.isVisibleInDev;
+
+    const classNames = isContentVisible
+      ? [
+          classes.content,
+          isEmpty ? classes.empty : '',
+          isPristine ? classes.pristine : '',
+        ].join(' ')
+      : '';
+
+    const [open, setOpen] = React.useState(false);
+    const devContent = isContentVisible && !isEmpty ? children : 'Dialog';
+
+    useEffect(() => {
+      B.defineFunction('handleDialogOpen', e => {
+        e.preventDefault();
+
+        setOpen(true);
+      });
+
+      B.defineFunction('handleDialogClose', e => {
+        e.preventDefault();
+
+        setOpen(false);
+      });
+    }, []);
+
+    return (
+      <div className={classNames}>
+        {isDev ? devContent : ''}
+
+        <Dialog open={open} onClose={() => setOpen(false)}>
+          <div className={classes.dialog}>{children}</div>
+        </Dialog>
+      </div>
+    );
+  })(),
+  styles: B => theme => {
+    const style = new B.Styling(theme);
+    const getSpacing = (idx, device = 'Mobile') =>
+      idx === '0' ? '0rem' : style.getSpacing(idx, device);
+    return {
+      dialog: {
+        backgroundColor: 'white',
+        margin: '0 auto',
+        padding: '1.5rem',
+        overflowWrap: 'break-word',
+      },
+      content: {
```

Comment:
This class is used on the DIV to hide the Dialog in. I don't see the use of padding etc. This has no effect on the shown Dialog.
Also, the extra DIV wrapper is something we need in the page builder to add the drag/drop events, but in runtime it has no purpose. So please add it with a condition unless you are sure that you need it in runtime.

Reply:
>This class is used on the DIV to hide the Dialog in. I don't see the use of padding etc. This has no effect on the shown Dialog.

Thanks for pointing that out. I'll be fixing it.


> Also, the extra DIV wrapper is something we need in the page builder to add the drag/drop events, but in runtime it has no purpose. So please add it with a condition unless you are sure that you need it in runtime.

I've tested this and it gave me no errors. Everything is working fine. I don't think I need an extra div, unless I'm missing something?